src/rajawali/Object3D.java

```diff
@@ -649,8 +649,8 @@ protected void cloneTo(Object3D clone, boolean copyMaterial) {
 	public Object3D clone(boolean copyMaterial, boolean cloneChildren) {
 		Object3D clone = new Object3D();
 		cloneTo(clone, copyMaterial);
-		clone.setRotation(getRotation());
-		clone.setScale(getScale());
+        clone.setRotation(getRotation().clone());
```

Comment:
This is redundant as the setRotation method calls setAll on the vector which copies the internal double values.

https://github.com/MasDennis/Rajawali/blob/master/src/rajawali/ATransformable3D.java#L212

Which then calls

https://github.com/MasDennis/Rajawali/blob/master/src/rajawali/math/vector/Vector3.java#L167